assets/src/edit-story/components/keyboard/index.js

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useContext, useEffect, createRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Context from './context';
+
+const INPUT_ELEMENTS = ['INPUT', 'TEXTAREA'];
+
+const globalRef = createRef();
+
+export function useKeyDownEffect(ref, keyNameOrSpec, callback, opts) {
+  const { keys } = useContext(Context);
+  useEffect(
+    () => {
+      const node = ref.current;
+      if (!node) {
+        return undefined;
+      }
+      const handler = createKeyHandler(keys, keyNameOrSpec, callback);
+      node.addEventListener('keydown', handler);
```

Comment:
Here's where mousetrap could be instantiated (as I mentioned it in my other comment).

Thoughts about using such a library for consistent keyboard handling? I remember @barklund exploring it.

Reply:
I'd be happy to switch to [mousetrap](https://www.npmjs.com/package/mousetrap) for the handler. And certainly text-based shortcut definition is good. But for some reason I was under the impression that we rejected Mousetrap for some reason? I remember there were some questions w.r.t. API on the other thread. The main part of this PR, imho, is the DOM-based layering.  Key handling is slightly less of a focus, but couple important pieces that we'd need to clarify with mousetrap:
 - Treatment of the `repeat: true/false`.
 - Treatment of input/non-input shortcuts.
 - Treatment of `shift` - in our world it's often convenient to handle the shortcut and allow either state of the `shift`, which I've been doing using `shift:'*'`.

I can dig in into mousetrap as an alternative for implementing key handler. So lmk if that's what we'd like to do.

Reply:
Gotcha! That makes sense. I think we can always keep mousetrap in mind and use it when really needed.

Reply:
I found a way to reconcile all of these concerns and switched to using Mousetrap now. PTAL.